frontend/src/scenes/surveys/Surveys.tsx

```diff
@@ -322,12 +330,15 @@ export function Surveys(): JSX.Element {
                                                             <LemonButton
                                                                 status="stealth"
                                                                 fullWidth
-                                                                onClick={() =>
+                                                                onClick={() => {
+                                                                    if (currentTeam && !currentTeam.surveys_opt_in) {
+                                                                        updateCurrentTeam({ surveys_opt_in: true })
+                                                                    }
                                                                     updateSurvey({
                                                                         id: survey.id,
                                                                         updatePayload: { end_date: null },
                                                                     })
-                                                                }
+                                                                }}
```

Comment:
this also seems to be handled in the logic (action for `resumeSurvey`) but maybe I'm misunderstanding. Does seem like this would be good to have in the logic if not.